api/files/serializers.py

```diff
@@ -421,15 +427,20 @@ def create(self, validated_data):
 
 class FileDetailSerializer(FileSerializer):
     """
-    Overrides FileSerializer to make id required.
+    - Overrides FileSerializer to make id required
+    - Files should return the id type they are queried with, but only in osfstorage is the id is reliably equivalent to
+     the path attribute, so that should not be overridden.
+
     """
-    id = IDField(source='_id', required=True)
+    id = GuidOrIDField(source='_id', required=True)
 
     def to_representation(self, value):
         data = super().to_representation(value)
-        guid = Guid.load(self.context['view'].kwargs['file_id'])
+        view = self.context['view']
+        guid = Guid.load(view.kwargs['file_id'])
         if guid:
             data['data']['id'] = guid._id
```

Comment:
Isn't this handled by the new `GuidOrIDField`?

Reply:
`GuidOrIDField` bypasses the validation that insists `_id` not be guid for `IDField` This actually changes the returned value. So basically `IDField` only effects POST/PATCH and this is for all returned JSON.

Reply:
fwiw, if the Guid lookup is actually required, then it makes sense to update the self link here, too, as that's otherwise populated by the `super().to_representation()` call. Putting it in the `if` makes it clear why we'd need to overwrite the default